docs/externalGrading.md

```diff
@@ -70,6 +70,8 @@ External grading configuration is done on a per-question basis. All configuratio
 
 * `enableNetworking`: Allows the container to access the public internet. This is disabled by default to make secure, isolated execution the default behavior.
 
+* `environment`: List of environment variables to set inside the grading container in the form accepted by the [Docker Engine API](https://docs.docker.com/engine/api/v1.41/#operation/ContainerCreate): `["VAR=value", ...]`. A variable without `=` is removed from the environment, rather than to have an empty value.
```

Comment:
IMO there's no reason to follow the pattern set by the Docker API - we're building something higher-level on top of that, so I think implementing as an object (key/value pair) makes more sense for a user-facing API. I spot-checked a few other projects and AWS Lambda, GCP Functions, Azure Functions, and Kubernetes all do (roughly) that instead of using `FOO=BAR`.

Reply:
Then I think it's a good choice to use the object, as it removes this ambiguity about the parsing without quotes.

Reply:
How should unsetting variables be handled?

Reply:
Sure, I can reformat. Thoughts on how (or whether) to support `unset`ting of variables? Docker will `unset FOO` given `["FOO"]`.

Maybe we can `unset` given `{"FOO": null}`, e.g.:

```js
Object.keys(env).map(key => env[key] === null ? key : `${key}=${env[key]}`)
```

Reply:
We'll need to make sure the key doesn't have spaces or other forbidden characters, perhaps:

```
> const env = {
... 'name': 'bob',
... 'age': 30,
... 'address': null,
... 'favorite food': 'green eggs and takoyaki'
... }
undefined
> Object.keys(env).map(key => env[key] === null ? key : `${key}=${env[key]}`)
[
  'name=bob',
  'age=30',
  'address',
  'favorite food=green eggs and takoyaki'
]
```